c2rust-analyze/src/rewrite/expr/mir_op.rs

```diff
@@ -383,6 +406,44 @@ impl<'a, 'tcx> ExprRewriteVisitor<'a, 'tcx> {
         (rewritten_ty, s)
     }
 
+    fn current_sub_loc_as_which_place(&self) -> Option<last_use::WhichPlace> {
+        // This logic is a bit imprecise, but should be correct in the cases where we actually call
+        // it (namely, when the `Place`/`Rvalue` is simply a `Local`).
+        let mut which = None;
+        for sl in &self.sub_loc {
+            match *sl {
+                SubLoc::Dest => {
+                    which = Some(last_use::WhichPlace::Lvalue);
```

Comment:
Can there be multiple `sub_loc`s that set `which` in this loop? Do you always want the last one?

If you only need one, you could exit early with `return`.

Reply:
Added a comment in 8f40e87fe91b83f96a642fc168c683edd253ecf1 - there can indeed be multiple matching `sub_loc`s, and we want the last one.